mypy/test/testcheck.py

```diff
@@ -217,6 +223,12 @@ def run_case_once(self, testcase: DataDrivenTestCase,
             assert sys.path[0] == plugin_dir
             del sys.path[0]
 
+        # when running the legacy tests in based mode, we just make sure they don't crash
+        if isinstance(self, BasedTypeCheckSuite):
+            if testcase.xfail:
+                raise AssertionError()
```

Comment:
this code doesn't seem to match the comment. if it's a based test then raise an error if it's an `xfail`?

Reply:
What it means is, after you run mypy, don't bother validating the output, if it's crashed it won't reach this line. if the test is xfail then fail it on purpose instead